lib/resources/bid_repository.dart

```diff
@@ -13,12 +13,17 @@ class BidRepository {
   BidClient _client;
   UserRepository _userRepository;
 
-  Future<List<Bid>> getMyBids() async {
-    return await _client.getMyBids();
-  }
-
   Future<Result<String>> placeBid(BuyerBidData data) async {
     final userId = await _userRepository.getProfileId();
     return await _client.placeBid(userId, data);
   }
+
+  Future<Result<List<Bid>>> getMyBids() async {
+    final userId = await _userRepository.getProfileId();
+    if (userId != null) {
```

Comment:
We probably don't need this null check. If we are making this call and we don't have userId, we should crash.

Reply:
okay